superdesk/publish/formatters/ninjs_ftp_formatter.py

```diff
@@ -47,48 +42,212 @@ def _transform_to_ninjs(self, article, subscriber, recursive=True):
         :param recursive:
         :return:
         """
-
+        # Get the path that the renditions will be pushed to
+        self.path = subscriber.get("destinations")[0].get("config").get("associated_path")
         include_original = subscriber.get("destinations")[0].get("config").get("include_original", False)
         if include_original:
             self.internal_renditions = ["original"]
 
-        ninjs = super()._transform_to_ninjs(article, subscriber, recursive)
+        formatted_article = deepcopy(article)
 
-        # Get the path that the renditions will be pushed to
-        path = subscriber.get("destinations")[0].get("config").get("associated_path")
+        if article.get("type") == "text" and recursive:
+            self.apply_product_filtering_to_associations(formatted_article, subscriber)
+
+        ninjs = super()._transform_to_ninjs(formatted_article, subscriber, recursive)
+
+        renditions = ninjs.get("renditions")
+        if renditions:
+            for name, rendition in renditions.items():
+                rendition["href"] = (
+                    self.path.lstrip("/")
+                    + ("/" if not self.path.endswith("/") and self.path else "")
+                    + get_rendition_file_name(rendition)
+                )
+
+        return ninjs
+
+    def apply_product_filtering_to_associations(self, article, subscriber):
```

Comment:
Until we have a way to configure this in the front-end, it would be good to have a system wide config to control this (and default to `False - do not filter associations`, so this becomes an opt-in feature as not to affect existing customers)

Reply:
Done!